clusterctl/clusterdeployer/clusterclient.go

```diff
@@ -207,6 +207,50 @@ func (c *clusterClient) CreateMachineObjects(machines []*clusterv1.Machine) erro
 	return nil
 }
 
+// DeleteApiServerStackDeployments attempts to delete apiserver stack resources
+func (c *clusterClient) DeleteApiServerStackDeployments() error {
+	// delete cluster controller
+	cmd := exec.Command("kubectl", c.buildKubectlDeleteDeploymentArgs("deployment","clusterapi-controllers")...)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		glog.V(2).Infof("running args kubeclt %v", c.buildKubectlDeleteDeploymentArgs("deployment","clusterapi-controllers"))
+		return fmt.Errorf("couldn't kubectl delete deployment clusterapi-controllers: %v, output: %s", err, string(out))
+	}
+
+	// delete apiserver
+	cmd = exec.Command("kubectl", c.buildKubectlDeleteDeploymentArgs("deployment","clusterapi-apiserver")...)
```

Comment:
Is there a reason why the name and form of apiserver and etcd is being hardcoded here? Why not kubectl delete the results of getApiServerYaml in cluster deployer?